src/Framework/Exceptions/Contracts/LoggableException.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace Give\Framework\Exceptions\Contracts;
+
+interface LoggableException {
```

Comment:
This seems like an unnecessary redundancy, if the `Loggable` trait is what is being used. The trait serves as a contract. It seems that we should have a preferred way instead of increasing the complexity.

The only application for a redundant interface that stands out is allowing a class to provide its own implementation. In which case, the class could override the trait methods.

Reply:
Yeah, I thought you might catch that. I really prefer Interfaces to signify contracts. I learned in the process of doing this that Traits can't implement Interfaces. I wanted the Interface to be the contract, and the Trait to be the helper. 

You can even see that when checking if a class is GiveWP's, I'm checking for both — this wasn't my intended behavior. I'm tempted to remove the trait check and add the Interface to the primitive exceptions and other native GiveWP exceptions.

What do you think?

Reply:
I was also considering removing the trait (`composition`) in favor of the interface (`inheritance`).

The interface is better for enforcing the contract and code re-use can be achieved with a base (or abstract) class.

The trait would come into play to separate the concerns of the implementation itself, if such separation were deemed necessary.

Reply:
The biggest reason for the Trait is the Primitive exceptions. They _have_ to extend the base PHP exception classes they extend. So I make a base `LoggedException` class for them to use. This also comes into play if someone wants to extend one of those classes, as well. I'm ok with thinking of the Interface as the contract and the Trait as a compositional helper.

Reply:
Tentative resolution in f83c8f6

Reply:
As a note, @kjohnson, what I'm introducing here _is compositional_ by nature. Whether an exception is loggable or not is pluggable, and shouldn't affect it's inheritance.

Reply:
Cool, yeah, I can get behind that 👍🏻